build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -418,11 +279,23 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
             }
 
             // cleanup compilers project output we don't need
-            DeleteMainPublishOutput(outputDir, "compilers");
+            PublishMutationUtilties.CleanPublishOutput(outputDir, "compilers");
             File.Delete(compilersDeps);
-            File.Delete(compilersRuntimeConfig);
+            
+            // Publish SharedFx
+            var sharedFrameworkNugetVersion = c.BuildContext.Get<string>("SharedFrameworkNugetVersion");
+            var commitHash = c.BuildContext.Get<string>("CommitHash");
+
+            var sharedFrameworkPublisher = new SharedFrameworkPublisher(
```

Comment:
No I agree that's how it would work. 

I hadn't yet put in the logic to pull the .zip in the cli-build so publishing to each stage still made sense. I agree that it should pull this and just copy it into `stage1` and `stage2`. 

Since we've come to conclusions on the repo structure, it makes sense to make that change immediately no, I think.